modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="troubleshooting-instrumentation_{context}"]
+= Troubleshooting instrumentation
+
+You can troubleshoot instrumentation issues as follows.
+
+.Procedure
+
+== Ensure the Instrumentation was injected properly in your workload
+
+. Check the instrumentation object is created:
++
+[source,console]
+----
+oc get instrumentation -n <your-project>
+----
+
+. Verify that the resources were deployed in the correct order
++
+Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.
+
+When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
+----
+
+You should see something like this:
+
+[source,console]
+----
+... Created container opentelemetry-auto-instrumentation
+... Started container opentelemetry-auto-instrumentation
+----
+
+If those messages are missing, then it means that was an issue while injecting the auto-instrumentation.
+
+. Inspect for the operator logs for instrumentation errors:
++
+[source,console]
+----
+oc logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n openshift-opentelemetry-operator --follow
+----
+
+. Confirm that the application pods you want to instrument are labeled correctly and the appropriate auto-instrumentation settings have been applied. For example, if you are instrumenting a Python application, you can check that the pods have the `instrumentation.opentelemetry.io/inject-python='true'` annotation:
++
+[source,console]
+----
+oc get pods -n <your-project> -o jsonpath='{range .items[?(@.metadata.annotations["instrumentation.opentelemetry.io/inject-python"]=="true")]}{.metadata.name}{"\n"}{end}'
+----
+
+You should double-check the annotation applied to the instrumentation object is the correct one for the language you are instrumenting. If there are multiple instrumentations in the same namespace, you should set its name in the annotation. Also, if the Instrumentation object is in a different namespace, you need to specify it in the annotation. You can find the accepted values in the link:otel-instrumentation.html["OpenTelemetry instrumentation configuration options"] section.
```

Comment:
Is this part of the step that starts on line 54?

Reply:
I would say yes.
Line 54 is about: use the correct annotation for the desired language instrumentation.
Line 61 is about: ok, you chose the correct annotation. Now, we need to check that you chose the correct value for that annotation.

Reply:
"If there are multiple instrumentations in the same namespace, you should set its [instrumentation's or namespaces'?] name in the annotation."

Reply:
There are 3 things you can place in the instrumentation annotation:
- `yes`: it will look for an Instrumentation object in the same namespace where the workload is running and will inject it into the workload.
- `<instrumentation instance name>`: will look for an Instrumentation object in the same namespace where the workload is running and will inject it into the workload.
- `<namespace>/<instrumentation instance name>`:  will look for an Instrumentation object in the specified namespace and will inject it into the workload.

I think this clarifies the sentence :D

Reply:
@iblancasa, I'm having trouble understanding this (I hope users will not). Would it be possible to add one example, just one YAML block that shows all of these three possibilities?

Reply:
Sure.

First case:
```yaml
instrumentation.opentelemetry.io/inject-nodejs: "true"
```

Second case:
```yaml
instrumentation.opentelemetry.io/inject-nodejs: "my-instrumentation"
```

Third-case:
```yaml
instrumentation.opentelemetry.io/inject-nodejs: "other-namespace/my-instrumentation"
```